pom.xml

```diff
@@ -204,6 +204,62 @@
         <excludedTests>com.google.cloud.spanner.pgadapter.IntegrationTest</excludedTests>
       </properties>
     </profile>
+
+    <profile>
+      <id>assembly</id>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-resources-plugin</artifactId>
+            <version>3.2.0</version>
```

Comment:
do we need this divergence between fat-jar and docker builds?
eventually this list may grow and the divergence always makes it hard to maintain and track

Reply:
This is a safer way to build an executable Java distribution than building a fat jar, and I would like to try to convert the fat-jar into one that uses a similar approach as this.

The potential problem with a fat jar is that it places everything into one jar, which means that if for example two different dependencies define the same configuration files or other resources, only one of the versions will be included in the fat jar. This can cause silent breakage if a dependency adds a new  file in a new version and expects it to be added to the final build. Example:

* dependency-1.jar contains `META-INF/services/io.grpc.LoadBalancerProvider` with the load balancer provider that it wants. This file will automatically be copied to `META-INF/services/io.grpc.LoadBalancerProvider` in the fat jar. The classes in dependency-1.jar will still be able to find the file, as it is in the expected location.
* During an update of the dependencies of PGAdapter dependency-2.jar also adds a `META-INF/services/io.grpc.LoadBalancerProvider`. The fat jar will only contain one of these files (and it is random which it is).

The setup for the Docker image keeps all dependency jars as they are and place these in a `lib` directory that is then automatically added to the classpath when PGAdapter is started.